### PR TITLE
Feat/url params

### PR DESCRIPTION
**Pull request by ShahAnuj2610**

### CODE REVIEW

**Comment by ShahAnuj2610** [comment]
Working GIF:
url_par

File: src/components/SearchBox.js [hunk 1]
```diff
@@ -205,6 +234,13 @@ class SearchBox extends Component {
       if (debounce > 0)
         this.searchBase.setValue(value, { triggerQuery: false });
       this.triggerSuggestionsQuery(value);
+      if (URLParams) {
+        window.history.replaceState(
```
Comment (jyash97) [inline src/components/SearchBox.js:238]: Will this work in SSR ?

File: src/components/SearchBox.js [hunk 2]
```diff
@@ -57,20 +59,45 @@ class SearchBox extends Component {
 
   componentDidMount() {
     this._initSearchBase();
+    if (this.props.URLParams) {
+      this.setValue({
+        value: this.getSearchTerm(this.props.currUrl),
```
Comment (jyash97) [inline src/components/SearchBox.js]: What is `currUrl` ? Is this the correct name ?

**Comment by jyash97** [comment]
Can you test this in different scenarios - 

- Reload the page when there is param
- Open the page on a new tab with the same param
- Use `<Link to="/route?param=value">` from `react-router-dom` 
- Use `history.push` from `react-router-dom`

File: src/components/SearchBox.js (hunk 2 above)
Comment (ShahAnuj2610) [reply src/components/SearchBox.js]: yes, it's the correct name. currUrl as in the URL in the browser tab

**Comment by ShahAnuj2610** [comment]
> * Reload the page when there is param

working

> * Open the page on a new tab with the same param

working

once, need to check with `react-router`

File: src/components/SearchBox.js (hunk 2 above)
Comment (jyash97) [reply src/components/SearchBox.js]: Shouldn't it be `currentURL` ? I don't think we will need to pass this from the component as a prop this should be tracked on lib side as we do this in Reactivesearch.

Comment (ShahAnuj2610) [reply src/components/SearchBox.js]: it's part of the library: https://github.com/appbaseio/react-searchbox/blob/a44ed57bf83fd73d22cc915a7f2efc9596cde623/src/index.js#L17.
The implementation is complex in RS, with the current implementation it's simple and I don't see any side-effects or anti-pattern

File: src/utils/helper.js [hunk 3]
```diff
@@ -47,3 +47,15 @@ export const debounce = (fn, delay) => {
...
+  const keyVal = {};
+  url.replace(/[?&]+([^=&]+)=([^&]*)/gi, function(m, key, value) {
+    key = decodeURIComponent(key);
+    value = decodeURIComponent(value);
+    keyVal[key] = value;
+  });
+  return keyVal;
+};
+
+export const isEmpty = val => val == null || !(Object.keys(val) || val).length;
```
Comment (jyash97) [inline src/utils/helper.js]: `val == null` will be true for `undefined` as well. 

I see this is been used at 2 places and the argument passed is of type `string` . So the later condition that is  `keys` and `length` is just for future use ?

Comment (ShahAnuj2610) [reply src/utils/helper.js]: > val == null will be true for undefined as well.

should return `true` for `undefined`

`(Object.keys(val)` => future
`!val.length` => this is for string, return true for `''`

Comment (jyash97) [reply src/utils/helper.js]: We can simplify this: 

```
!(val && val.length && Object.keys(val).length)
```

Wdyt ?

Comment (ShahAnuj2610) [reply src/utils/helper.js]: cool

**Review by jyash97 — APPROVED** [review]
Awesome 🎉

**Pull request merged**